src/modules/siteRDF/process.cpp

```diff
@@ -36,15 +38,23 @@ Module::ExecutionResult SiteRDFModule::process(ModuleContext &moduleContext)
         histAB.initialise(distanceRange_.x, distanceRange_.y, distanceRange_.z);
     histAB.zeroBins();
 
-    for (const auto &[siteA, indexA] : a.sites())
-    {
-        for (const auto &[siteB, indexB] : b.sites())
-        {
-            if (excludeSameMolecule_ && (siteB->molecule() == siteA->molecule()))
-                continue;
-            histAB.bin(targetConfiguration_->box()->minimumDistance(siteA->origin(), siteB->origin()));
-        }
-    }
+    auto combinableHistograms = dissolve::CombinableValue<Histogram1D>([&histAB]() { return histAB; });
+
+    dissolve::for_each(std::execution::seq, a.sites().begin(), a.sites().end(),
```

Comment:
I'm guessing for performance reasons we probably want to change this :smile:
```suggestion
    dissolve::for_each(std::execution::par_seq, a.sites().begin(), a.sites().end(),
```